### PR TITLE
display line numbers for test programs

### PR DESCRIPTION
**Pull request by ono-max**
Displaying line numbers in test programs is easy to write test code.

### Reference

This method is based on [byebug ](https://github.com/deivid-rodriguez/byebug)project

### CODE REVIEW

File: test/support/utils.rb [hunk 1]
```diff
@@ -75,5 +75,9 @@ def create_pseudo_terminal
       p "terminal finished without reading 'y'"
       raise
     end
+
+    def strip_line_num(str)
+      str.gsub(/ *\d+\| ?/, '')
```
Comment (olleolleolle) [inline test/support/utils.rb:80]: Perhaps an anchor in the regular expression would help?

```suggestion
      str.gsub(/^\s*\d+\| ?/, '')
```

Comment (ono-max) [reply test/support/utils.rb:80]: Yes, you are right. Thanks!

**Comment by st0012** [comment]
I think it's a great idea to have line number 👍 
But since it needs to be typed manually, I feel `:` is easier to type than `|` (no need to move my pinky finger that much). Just my personal preference though 🙂

File: test/debug/break_test.rb [hunk 2]
```diff
@@ -8,29 +8,29 @@ module DEBUGGER__
...
-            def self.c
-              1
-            end
-          end
-          Bar.a
-          bar = Bar.new
-          bar.b(1)
-          Baz.c
-        end
+      strip_line_num <<~RUBY
```
Comment (st0012) [inline test/debug/break_test.rb:11]: for consistency reason, we may want **all** program code to have line numbers. and if that makes sense to you, maybe we can strip the line at a single place (perhaps in `debug_code`)? then we won't need to call this in every `program` method.

**Comment by ko1** [comment]
> But since it needs to be typed manually, I feel `:` is easier to type than `|` (no need to move my pinky finger that much). Just my personal preference though 🙂

BTW do you prefer `:` for `list` command?
pry, irb uses `:` instead of `|`.

**Comment by st0012** [comment]
> > But since it needs to be typed manually, I feel `:` is easier to type than `|` (no need to move my pinky finger that much). Just my personal preference though 🙂
> 
> BTW do you prefer `:` for `list` command?
> pry, irb uses `:` instead of `|`.

in that case I actually prefer `|`. but I'm ok with `:` for consistency reason as well.

File: test/debug/break_test.rb (hunk 2 above)
Comment (ono-max) [reply test/debug/break_test.rb]: I agree with your opinion. Thank you!
https://github.com/ruby/debug/pull/42/files#diff-ee14bf65f14927f7778a246f90c5cec0c581be3e2658c2a8a3726c59db5be3f9R22

**Pull request merged**